controllers/etcdcluster_controller.go

```diff
@@ -0,0 +1,82 @@
+package controllers
+
+import (
+	"context"
+	v1 "k8s.io/api/core/v1"
+	"time"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
+)
+
+// EtcdClusterReconciler reconciles a EtcdCluster object
+type EtcdClusterReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters,verbs=get;list;watch;create;update;patch;delete
```

Comment:
This controller doesn't need permission to create/update/patch/delete the `etcdclusters` resource